mapbox_gl_platform_interface/lib/src/view_wrappers.dart

```diff
@@ -0,0 +1,284 @@
+part of mapbox_gl_platform_interface;
+
+/// This file wrapps AndroidViewController classes in order to delay disposal process.
+/// It is an workaround for flutter 3, where resourses get disposed quicker than before, while Mapbox behaves badly
+/// and tries to access those resources after they had been disposed, resulting in a native crash.
+
+class WrappedPlatformViewsService {  
+  static AndroidViewController initAndroidView({
+    required int id,
+    required String viewType,
+    required TextDirection layoutDirection,
+    dynamic creationParams,
+    MessageCodec<dynamic>? creationParamsCodec,
+    VoidCallback? onFocus,
+  }) {    
+    final view = PlatformViewsService.initAndroidView(
+      id: id, 
+      viewType: viewType, 
+      layoutDirection: layoutDirection, 
+      creationParams: creationParams,
+      creationParamsCodec: creationParamsCodec,
+      onFocus: onFocus,
+    );
+    return TextureAndroidViewControllerWrapper(view as TextureAndroidViewController);
+  }
+}
+
+class TextureAndroidViewControllerWrapper implements TextureAndroidViewController {
+  TextureAndroidViewControllerWrapper(this._controller);
+
+  final TextureAndroidViewController _controller;
+
+  @override
+  PointTransformer get pointTransformer => _controller.pointTransformer;
+  set pointTransformer(PointTransformer transformer) => _controller.pointTransformer = transformer;
+
+  @override
+  void addOnPlatformViewCreatedListener(PlatformViewCreatedCallback listener) => 
+    _controller.addOnPlatformViewCreatedListener(listener);
+
+  @override
+  bool get awaitingCreation => _controller.awaitingCreation;
+
+  @override
+  Future<void> clearFocus() => _controller.clearFocus();
+
+  @override
+  Future<void> create({Size? size}) => _controller.create(size: size);
+
+  @override
+  // ignore: invalid_use_of_visible_for_testing_member
+  List<PlatformViewCreatedCallback> get createdCallbacks => _controller.createdCallbacks;
+
+  @override
+  Future<void> dispatchPointerEvent(PointerEvent event) => _controller.dispatchPointerEvent(event);
+
+  @override
+  //! workaround for flutter 3.0
+  Future<void> dispose() {
```

Comment:
Could you not just extend TextureAndroidViewController and only override `dispose `?

Reply:
thinking about this further - you need to use composition here to reuse the existing object - all is good